gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -496,6 +501,7 @@ function roleselection.GetSelectableRolesList(maxPlys, rolesAmountList)
 					end
 
 					subrole = cleanedLayerTbl[math.random(#cleanedLayerTbl)]
+					table.RemoveByValue(currentSubroleTbl, subrole)
```

Comment:
Looking at this now, I'm curious if we're negating the ability to have a sub-role multiple times. For instance, if a particular sub-role has been configured to have up to 2 per round, by only adding it once to selectableRoles and then removing it from the master sub role list, are we preventing it from later being able to assign it twice? It looks like SetSubRoles will remove it in that case.

Reply:
All lists in roleselection.GetSelectableRolesList are local, at line 518 is the important where the number of roles that can be chose get inserted.